src/main/scala/scorex/core/newserialization/ByteStringWriter.scala

```diff
@@ -0,0 +1,128 @@
+package scorex.core.newserialization
+
+import java.nio.ByteOrder
+import java.util.{Arrays, BitSet}
+
+import akka.util.ByteString
+
+class ByteStringWriter extends ScorexWriter {
+  override type CH = ByteString
+  private implicit val byteOrder = ByteOrder.BIG_ENDIAN
+
+  override def newWriter(): ScorexWriter.Aux[CH] = {
+    new ByteStringWriter()
+  }
+
+  private val builder = ByteString.createBuilder
+
+  override def length(): Int = builder.length
+
+  override def putChunk(byteString: ByteString): this.type = {
+    builder.append(byteString)
+    this
+  }
+
+  override def putByteString2(byteString: ByteString): this.type = {
+    putChunk(byteString)
+  }
+
+  override def put(x: Byte): this.type = {
+    builder.putByte(x)
+    this
+  }
+
+  override def putBoolean(x: Boolean): this.type = {
+    val byte:Byte = if (x) 0x01 else 0x00
+    builder.putByte(byte)
+    this
+  }
+
+  override def putShort(x: Short): this.type = {
+    builder.putShort(x)
+    this
+  }
+
+  /**
+    * Encode Short that are positive
+    *
+    * Use [[putShort]] to encode values that might be negative.
+    *
+    * @param x Short
+    */
+  override def putUShort(x: Int): this.type = {
+    assert(x >= 0 && x <= 0xFFFF, s"$x is out of unsigned short range")
+    builder.putShort(x.toShort)
+    this
+  }
+
+  /**
+    * Encode signed Int.
+    * Use [[putUInt]] to encode values that are positive.
+    *
+    * @param x Int
+    */
+  override def putInt(x: Int): this.type = {
+    builder.putInt(x)
+    this
+  }
+
+  /**
+    * Encode Int that are positive.
+    * Use [[putInt]] to encode values that might be negative.
+    * @param x Int
+    */
+  override def putUInt(x: Long): this.type = {
```

Comment:
@VictorMikheev Why not use VLQ encoding with `builder.putBytes`?

Reply:
This will change the protocol.